elevate-project/controllers/project.js

```diff
@@ -1,29 +1,87 @@
+/**
+ * name : controllers/project.js
+ * author : Adithya Dinesh
+ * Date : 22-Aug-2024
+ * Description : Orchestration controller for project
+ */
+
 const routeConfigs = require('../constants/routes')
 const requesters = require('../utils/requester')
-const requestParser = require('../utils/requestParser')
-
-// const createUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body)
-// }
-// const updateUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-
-// 	//const filteredRequestBody = requestParser.transformUpdateUserBody(req.body)
-// 	return await requesters.patch(req.baseUrl, selectedConfig.targetRoute.path, req.body, req.headers)
-// }
-
-// const entityTypeRead = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body, {
-// 		'X-auth-token': req.headers['x-auth-token'],
-// 	})
-// }
-
-const userController = {
-	// createUser,
-	// updateUser,
-	// entityTypeRead,
+const common = require('../constants/common')
+/**
+ * Fetch project templates from projects service.
+ * @name fetchProjectTemplates
+ * @param {Object} req
+ * @param {Object} res
+ * @param {Object} responses  
+ * @returns {JSON} - List of project templates
+ */
+/**
+ * req.body = {
+ * 		organization_id : 1,
+ * 		resourceType : ['survey','projects']
+ * }
+ */
+
+const fetchProjectTemplates = async (req, res, responses) => {
+	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
+
+	let response = { result: { data: [], count: 0 } }
+	let proceedToCallProjectService = true
+	let resp = {}
+
+	// fetch the max limit from the env file for the DB Find API
+	const max_limit = process.env.RESOURCE_MAX_FETCH_LIMIT ? parseInt(process.env.RESOURCE_MAX_FETCH_LIMIT, 10) : 1000
+
+	if (req.body) {
+		const resourceType = req?.body?.resourceType || [];
```

Comment:
@adithyadinesh0412 if req body is there and resource type is valid then make the key as true

Reply:
No , we cannot proceed if we have resourceType. Because if there are resource type not relevant to project service like survey , obs etc... we need not call the api @priyanka-TL